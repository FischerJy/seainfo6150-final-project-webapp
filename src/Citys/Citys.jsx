import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";

var list = ["London", "Lobeira", "Cudillero", "Colunga", "Somiedo", "Shanghai"];

export default function Citys() {
  return (
    <div className="city-list" style={{ textAlign: "center" }}>
      {list.map((it, index) => (
        <Link to={`/city/${it}`} key={index}>
          {it}
        </Link>
      ))}
    </div>
  );
}
