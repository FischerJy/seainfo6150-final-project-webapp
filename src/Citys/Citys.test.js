import React from "react";
import { render } from "@testing-library/react";
import Citys from "./Citys.jsx";
import { BrowserRouter } from "react-router-dom";



it("renders the Citys component correctly", () => {
    const { container } = render(<BrowserRouter>
        <Citys />
    </BrowserRouter>);
    expect(container).toMatchSnapshot();
});