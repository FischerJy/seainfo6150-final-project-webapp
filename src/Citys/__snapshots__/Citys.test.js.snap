// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Citys component correctly 1`] = `
<div>
  <div
    class="city-list"
    style="text-align: center;"
  >
    <a
      href="/city/London"
    >
      London
    </a>
    <a
      href="/city/Lobeira"
    >
      Lobeira
    </a>
    <a
      href="/city/Cudillero"
    >
      Cudillero
    </a>
    <a
      href="/city/Colunga"
    >
      Colunga
    </a>
    <a
      href="/city/Somiedo"
    >
      Somiedo
    </a>
    <a
      href="/city/Shanghai"
    >
      Shanghai
    </a>
  </div>
</div>
`;
